### DISCUSSION TITLE
[@types/k6] chromium is not available on k6/experimental/browser

### DISCUSSION THREAD

**Discussion by ArCiGo** [DefinitelyTyped/DefinitelyTyped#65813] [Issues with a @types package]
`chromium` module is not available on `k6/experimental/browser` for `@types/k6`. It's needed for [k6 Browser](https://k6.io/blog/get-started-with-k6-browser/) feature used with Playwright.

[a comment by an automated account is not shown]

**Reply by ankur22** [reply]
@ArCiGo, Thanks for raising this. Which version of k6 are you using? The latest released version of k6 ([v0.45.0](https://github.com/grafana/k6/releases/tag/v0.45.0)) should still work with the current type definitions.

If you are working with a custom build from the `main` branch of the [browser module](https://github.com/grafana/xk6-browser), then you will find that `chromium` is no longer available in `k6/experimental/browser`. We will have more updates around this and other related changes in the following release of k6 (v0.46.0). If you are keen to get a head start, check out the [examples](https://github.com/grafana/xk6-browser/tree/main/examples) to get more of an idea of what changes have been made already.

Hope this helps!

**Reply by ankur22** [reply]
One thing to add to this is that, due to the upcoming change where `chromium` will no longer be exported, we're unlikely to make the change to add the `chromium` type definition. We will endeavour to have the correct type definitions ready for the upcoming release though.

**Reply by ArCiGo** [reply]
Hi, @ankur22 . About **k6**: I was using the version _0.44.3_ of `@types/k6` and version _0.44.1_ of k6 (for Mac). Both were installed on June 16th. About **Playwright** (just in case), I'm using the version `1.34.0` of `@playwright/test`.

I updated both, and I was following this [sample](https://github.com/grafana/xk6-browser/blob/main/examples/locator_pom.js), but I found that the @types/k6 doesn't have the browser member at `k6/experimental/browser` neither `k6/x/browser` (this one doesn't exist). Am I missing something? Or how can I implement it?

On the other hand, following the sample I mentioned, will it solve what I try to do with my current solution using Playwright?

This I a sample of [what I want to do](https://community.k6.io/t/typescript-example/449/6?u=arcigo).-

```typescript
import { sleep, check } from "k6";
import { chromium } from 'k6/experimental/browser';

import { SLLoginPage } from '../pages/SLLoginPage';
import { SLProductPage } from '../pages/SLProductsPage';

let slLoginPage: SLLoginPage;
let slProductPage: SLProductPage;

export const options = {
    scenarios: {
        login: {
            executor: 'constant-vus',
            exec: 'login',
            vus: 1
        }
    },
};

export async function login() {
    const browser = chromium.launch({
        headless: true,
        slowMo: '500ms'
    });
    const context = browser.newContext();
    const page = context.newPage();

    try { 
        slLoginPage = new SLLoginPage(page);
        slProductPage = new SLProductPage(page);

        await slLoginPage.goTo();
        await slLoginPage.fillLoginForm(process.env.USERNAME as string, process.env.PASSWORD as string);
        await page.waitForURL('**/dashboard', { timeout: 3000 });
} finally {
    page.close();
    browser.close();
}
```

**Reply by ArCiGo** [reply]
@ankur22 Are there any updates about how to use the browser member to implement it with an existing POM using Playwright?

**Comment by ankur22** [comment]
> I updated both, and I was following this [sample](https://github.com/grafana/xk6-browser/blob/main/examples/locator_pom.js), but I found that the @types/k6 doesn't have the browser member at k6/experimental/browser neither k6/x/browser (this one doesn't exist). Am I missing something? Or how can I implement it?

@ArCiGo, the `browser` member is not part of the type definitions, that's correct. This is coming in an upcoming release and update to the type definitions for the browser module in k6. It's worth reiterating that `chromium` will be removed from that same release and type definition update.

This [sample](https://github.com/grafana/xk6-browser/blob/main/examples/locator_pom.js) you are working with only works with the HEAD of `main`. Try working with this [v0.10.0 tagged sample](https://github.com/grafana/xk6-browser/blob/v0.10.0/examples/locator_pom.js) instead.

I hope that the new link to the examples will help you resolve the issues your are facing. Any specific questions on why the test isn't working, are best asked in our [browser module community forum](https://community.k6.io/c/k6-browser/14).
